Source/ACE.Server/WorldObjects/Player_Magic.cs

```diff
@@ -1133,8 +1134,9 @@ public bool CreatePlayerSpell(uint spellId)
             if (spell == null)
                 return false;
 
-            // get player's current magic skill
-            var magicSkill = GetCreatureSkill(spell.School).Current;
+            // get player's current magic skill (matches CACQualities::InqSkillLevel)
+            var playerSkill = GetCreatureSkill(spell.School);
+            var magicSkill = playerSkill.InitLevel + playerSkill.Ranks;
```

Comment:
You can't change magicSkill here. This effectively means a player's buffs will be useless for all magic calculations from this point forward.